File: backend/src/main/java/com/example/demo/Entity/R_TD.java
package com.example.demo.Entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.util.Date;

@Entity
@Setter
@Getter
@Table(uniqueConstraints = {@UniqueConstraint(columnNames = {"turno_lavorativo_date","dipendente_id","turno_lavorativo_id"})} )
public class R_TD {

    @Id
    @GeneratedValue
    private long id;

    @ManyToOne
    @JoinColumn
    private Dipendente dipendente;

    @ManyToOne
    @JoinColumn
    private TurnoLavorativo turnoLavorativo;

    private boolean straordinario;

    @Temporal(TemporalType.DATE)
    @Column(name = "turno_lavorativo_date")
    private Date turnoLavorativoDate;

}

File: frontend/src/app/model/R_TD.ts
import { Dipendente } from "./Dipendente";
import { TurnoLavorativo } from "./TurnoLavorativo";

export interface R_TD{

    id:number;
    dipendente:Dipendente;
    turnoLavorativo:TurnoLavorativo;
    straordinario:boolean;
    turnoLavorativoDate:Date;

}

export class R_TD{

    id!:number;
    dipendente:Dipendente;
    turnoLavorativo:TurnoLavorativo;
    straordinario:boolean;
    turnoLavorativoDate:Date;

    constructor(dip:Dipendente, turno:TurnoLavorativo, s:boolean, d:Date){
        this.dipendente = this.dipendente;
        this.turnoLavorativo = turno;
        this.straordinario = this.straordinario;
        this.turnoLavorativoDate = d;
    }

}